enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all(*args)
+    my_each do |x|
+      return true unless block_given?
+
+      return false unless yield(x)
+    end
+    true
+  end
+
+  def my_any?(*args)
+    args = nil
+    if args
+      my_each { |x| return false unless x == args }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if x == yield(x) }
+    end
+    false
+  end
+
+  def my_none(*args)
+    !my_any?
+  end
```

Comment:
The issues with **my_none?**, **my_any?** and **my_all?** are similar. the methods should be able to: 

* [x] Handle the cases where no block is given (Hint: block_given?)
* [ ] Return a suitable boolean result if no arguments were passed.
  ```ruby
	array = [false, true, false, nil, []]
	puts array.my_any? # should print true
	puts array.my_all? # should print false
	puts array.my_none? # should print false
  ```
* [ ] Accept primitives, classes (i.e Integer, String, etc...) and Regex Expressions as arguments.
  ```ruby
	words = %w[dog door rod blade]
	puts words.my_all?(/c/) == words.all?(/c/) # false
	puts words.my_any?(/c/) == words.all?(/c/) # false
	puts words.my_none?(/c/) == words.all?(/c/) # true
  ```